core/domain/user_jobs_one_off.py

```diff
@@ -439,3 +439,43 @@ def map(model_instance):
     @staticmethod
     def reduce(key, values):
         yield (key, len(values))
+
+
+class FixUserSettingCreatedOnOneOffJob(
+        jobs.BaseMapReduceOneOffJobManager):
+    """One off job that updates the created_on field by finding the minimum
+     date-time among other fields.
+    """
+
+    _USER_PROCESSED = 'user_processed'
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        """Return a list of datastore class references to map over."""
+
+        return [user_models.UserSettingsModel]
+
+    @staticmethod
+    def map(model_instance):
+        if not model_instance.deleted:
+            user_dates = [
+                model_instance.last_agreed_to_terms,
+                model_instance.last_started_state_editor_tutorial,
+                model_instance.last_started_state_translation_tutorial,
+                model_instance.last_logged_in,
+                model_instance.last_edited_an_exploration,
+                model_instance.last_created_an_exploration]
+
+            # Removing None values from the user_dates list.
+            filtered_user_dates_list = list(filter(None, user_dates))
```

Comment:
```suggestion
            filtered_user_dates_list = list(filter(lambda date: date is not None, user_dates)
```

Reply:
`None` in the first argument filters values that are `True` which is not exactly what we want.